include/arch/x86/intel64/linker.ld

```diff
@@ -107,15 +107,16 @@ SECTIONS
 	*(.rodata)
 	*(.rodata.*)
 
-	#include <snippets-rodata.ld>
-
 #ifdef CONFIG_X86_MMU
 	. = ALIGN(8);
 	_mmu_region_list_start = .;
 	KEEP(*("._mmu_region.static.*"))
 	_mmu_region_list_end = .;
 #endif /* CONFIG_X86_MMU */
 
+/* Move this section after mmu region in order to fix error for running GCOV.*/
+#include <snippets-rodata.ld>
+
```

Comment:
Would this be more appropriate as a separate commit? It resolves an error for running GCOV for any x86 with MMU config, not specific to enabling code coverage for qemu_x86_64. Or am I mistaken?

Reply:
yes the linker.ld changes need to be their own commit

Reply:
@jenmwms @andrewboie  This change is also a part of enable enabling code coverage for qemu_x86_64. And I will split them into 2 commit, thank for your comment~